src/main/scala/ai/privado/languageEngine/java/passes/config/JavaPropertyLinkerPass.scala

```diff
@@ -61,30 +62,50 @@ class JavaPropertyLinkerPass(cpg: Cpg) extends PrivadoParallelCpgPass[JavaProper
     val membersAndValues = annotatedMembers()
 
     membersAndValues
-      .filter { case (key, _) => propertyNode.name == key.code.slice(3, key.code.length - 2) }
+      .filter { case (key, _) => propertyNode.name == Option(key.code.slice(3, key.code.length - 2)).getOrElse("") }
+      .foreach { case (_, value) =>
+        builder.addEdge(propertyNode, value, EdgeTypes.IS_USED_AT)
+        builder.addEdge(value, propertyNode, EdgeTypes.ORIGINAL_PROPERTY)
+      }
+
+    val annotatedMethodsList = annotatedMethods()
+
+    annotatedMethodsList
+      .filter { case (key, _) =>
+        propertyNode.name == Option(key.code.slice(3, key.code.length - 2)).getOrElse("")
+      }
       .foreach { case (_, value) =>
         builder.addEdge(propertyNode, value, EdgeTypes.IS_USED_AT)
         builder.addEdge(value, propertyNode, EdgeTypes.ORIGINAL_PROPERTY)
       }
   }
 
+  /** List of all methods annotated with Spring's `Value` annotation
+    */
+  private def annotatedMethods(): List[(AnnotationParameterAssign, Method)] = cpg.annotation
+    .name(".*Value.*")
```

Comment:
If we are checking on name, do exact match instead to avoid FP `.nameExact("Value")`